python/ray/data/_internal/stats.py

```diff
@@ -898,22 +902,29 @@ def from_block_metadata(
                 if rounded_total <= 0:
                     # Handle -0.0 case.
                     rounded_total = 0
-                exec_summary_str = "{}/{} blocks executed in {}s".format(
-                    len(exec_stats), len(block_metas), rounded_total
+                exec_summary_str = "{} blocks produced in {}s".format(
+                    len(exec_stats), rounded_total
                 )
             else:
                 exec_summary_str = ""
-            if len(exec_stats) < len(block_metas):
-                if exec_stats:
-                    exec_summary_str += ", "
-                num_inherited = len(block_metas) - len(exec_stats)
-                exec_summary_str += "{}/{} blocks split from parent".format(
-                    num_inherited, len(block_metas)
-                )
-                if not exec_stats:
-                    exec_summary_str += " in {}s".format(rounded_total)
             exec_summary_str += "\n"
 
+        task_rows = collections.defaultdict(int)
+        for meta in block_metas:
+            if meta.num_rows is not None and meta.exec_stats is not None:
+                task_rows[meta.exec_stats.task_idx] += meta.num_rows
+        task_rows_stats = None
+        if len(task_rows) > 0:
+            task_rows_stats = {
+                "min": min(task_rows.values()),
+                "max": max(task_rows.values()),
+                "mean": int(np.mean(list(task_rows.values()))),
+                "count": len(task_rows),
+            }
+            exec_summary_str = "{} tasks executed, {}".format(
+                len(task_rows), exec_summary_str
+            )
```

Comment:
just to confirm, this is new code unrelated to the deprecation, right?

Reply:
Yeah this part is for the issue I linked in the description, the rest is just renaming